go/array_string/permutation/permutation.go

```diff
@@ -0,0 +1,29 @@
+// Package permutation has the implementation of operations
+// that has to do do it with strings and permutations
+package permutation
+
+func checkPermutation(inputA, inputB string) bool {
+	// validation
+	if len(inputA) != len(inputB) {
+		return false
+	}
+
+	inputACharSet := make(map[rune]int)
+
+	for _, char := range inputA {
+		inputACharSet[char]++
+	}
+
+	for _, char := range inputB {
+
+		if _, ok := inputACharSet[char]; ok {
```

Comment:
`if n, ok := … {` because you use `n` again on the next line to check if it's zero.

Reply:
Using the earlier suggested approach will not require this, I think.